code/modules/atmospherics/machinery/components/unary_devices/cryo.dm

```diff
@@ -254,17 +263,16 @@ GLOBAL_VAR_INIT(cryo_overlay_cover_off, mutable_appearance('icons/obj/cryogenics
 
 	var/datum/gas_mixture/air1 = airs[1]
 
-	if(air1.gases.len)
+	if(air1.gases[/datum/gas/oxygen][MOLES] > CRYO_MIN_GAS_MOLES)
 		if(mob_occupant.bodytemperature < T0C) // Sleepytime. Why? More cryo magic.
 			mob_occupant.Sleeping((mob_occupant.bodytemperature * sleep_factor) * 1000 * delta_time)
 			mob_occupant.Unconscious((mob_occupant.bodytemperature * unconscious_factor) * 1000 * delta_time)
 		if(beaker)
 			if(reagent_transfer == 0) // Magically transfer reagents. Because cryo magic.
-				beaker.reagents.trans_to(occupant, 1, efficiency * 12.5 * delta_time, methods = VAPOR) // Transfer reagents.
-				air1.gases[/datum/gas/oxygen][MOLES] -= max(0, air1.gases[/datum/gas/oxygen][MOLES] - delta_time / efficiency) //Let's use gas for this
-				air1.garbage_collect()
-			reagent_transfer += 0.5 * delta_time
-			if(reagent_transfer >= 10 * efficiency) // Throttle reagent transfer (higher efficiency will transfer the same amount but consume less from the beaker).
+				beaker.reagents.trans_to(occupant, CRYO_TX_QTY * delta_time, efficiency * CRYO_MULTIPLY_FACTOR, methods = VAPOR) // Transfer reagents.
+				consume_gas += delta_time // Keep track of how many transfers we do so we can consume gas in process_atmos()
```

Comment:
In high lag situations this will cause large dumps of gas at once when atmos finally processes won't it? Could we do anything to prevent this?

Reply:
It could just consume a static amount of gas while active, instead of consuming that, plus the extra each time it transfers reagent. The static consumption already makes up the the majority of gas used.

Reply:
Maybe something like a binary "have we transferred anything" instead of additive delta time?

Reply:
Works for me, and it's just as good provided machines and atmos don't get too far out of sync. I'll push it today.